modules/lang-painless/src/main/java/org/elasticsearch/painless/MethodWriter.java

```diff
@@ -125,7 +125,8 @@ public void writeStatementOffset(Location location) {
         int offset = location.getOffset();
         // ensure we don't have duplicate stuff going in here. can catch bugs
         // (e.g. nodes get assigned wrong offsets by antlr walker)
-        assert statements.get(offset) == false;
+        // TODO: introduce a way to ignore internal statements so this assert is not triggered
```

Comment:
Consider cutting an issue for TODOs like this so they don't get lost, then ref the issue in the comment.

Reply:
Done: https://github.com/elastic/elasticsearch/issues/51836

Reply:
Pls reference from TODO.